app/controllers/concerns/application_shared.rb

```diff
@@ -33,7 +33,9 @@ def set_taxonomy
           if params["#{taxonomy}_id"].blank? # the key is present and explicitly set to nil which indicates "any" context
             determined_taxonomy = nil
           else
-            determined_taxonomy = available.where(:id => params["#{taxonomy}_id"]).first
+            identifier = params["#{taxonomy}_id"]
+            determined_taxonomy = available.friendly.find(identifier) rescue ActiveRecord::RecordNotFound
+            determined_taxonomy ||= available.where(:id => identifier).first
```

Comment:
the tests (or at least some) were failing because the new didn't address the case, where id was in for of "123-name", rails auto converts that to 123, since id is integer. Friendly find does not work with it. So we fallback to old way if friendly id didn't find anything.

Reply:
shouldn't we fix the incorrect tests instead?

Reply:
well, but 123-name is generally accepted id form too, or am I missing something?

Reply:
Only where we use [`Parameterizable::ByIdName`](https://github.com/theforeman/foreman/blob/develop/app/models/concerns/parameterizable.rb#L20) (and taxonomies friendly id is by title).
tbh, we're quite inconsistent about these - some places use friendly, some places use parameterizable, and each has several variants.

Reply:
or actually, we do include that in taxonomies as well. interesting...

Reply:
this could probably be resolved properly by using the `scope_by_resource_id` method introduced by https://github.com/theforeman/foreman/pull/7289/

Reply:
testing that, let's see if that makes tests happy